### ISSUE TITLE
KLDivLoss should not have the 1/n coefficient, according to Wikipedia

### ISSUE THREAD

**Issue by cai-lw** [pytorch/pytorch#2259]
https://en.wikipedia.org/wiki/Kullback–Leibler_divergence#Definition

Also, I found the current form of KLDivLoss not very useful. In my opinion, it should resemble the form of other probability-based loss functions, that is, it should have a signature like (N,C)x(N,C)->(N,).

cc @fritzo @neerajprad @alicanb @vishwakftw

**Comment by soumith** [comment]
@jekbradbury and @moustapha6c what do you think about this suggestion?

**Comment by jekbradbury** [comment]
All the loss functions should ideally (optionally) return a vector that hasn’t been averaged over the batch. But the other change in behavior would be an API switch that doesn’t really change what’s possible but does break existing code.

**Comment by hughperkins** [comment]
Agree with t-vi. Simply add a parameter taht defaults to current behavior.  At least make it clear in the doc why the output of kldivloss doesnt match the output if one calculates kldiv by hand :)

**Comment by fritzo** [comment]
I think this was mislabeled module:distributions. Does anyone know the correct module?

**Comment by gchanan** [comment]
Given we have the `reduction` parameter  now, I'm closing this.

**Comment by t-vi** [comment]
I think this is about the "mean" reduction mode, which is documented to be changing behaviour with the next major release.
